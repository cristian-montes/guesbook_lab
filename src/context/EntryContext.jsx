import { createContext, useContext } from "react";
import { useState } from "react/cjs/react.development";

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
    const [entries, setEntries] = useState([]);

    return <EntryContext.Provider value={{entries, setEntries}}> { children } </EntryContext.Provider>

}

const useEntries = () => {
    const context = useContext(EntryContext);

    if(context === undefined){
        throw Error('useEntries must be used within an EntryContext Provider');
    }

    return context
}

export { EntryProvider, useEntries}
