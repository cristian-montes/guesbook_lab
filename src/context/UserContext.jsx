import React from "react";
import { createContext, useContext } from "react";
import { useState } from "react/cjs/react.development";

const UserContext  = createContext();

const UserProvider = ({ children }) => {
    const [user, setUser] = useState('')
    return <UserContext.Provider value={{user, setUser}}> {children} </UserContext.Provider>
}

const useUser = () => {
    const context = useContext(UserContext);

    if(context === undefined){
        throw new Error('useUser hook must be called within a UserContext Provider')
    }

    return context;
}

export { UserProvider, useUser}