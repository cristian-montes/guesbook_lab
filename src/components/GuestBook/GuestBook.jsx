import { useState } from "react";
import { useUser } from "../../context/UserContext";

export default function GuestBook(){
    const [ name, setName] = useState('');
    const [guesEntry, setGuesEntry] = useState('');
    // const { entries, setEntries } = useEntries();
    const { user, setUser } = useUser();

    return(
        <>
            <h1> Display message will go here</h1>
            <form>
                <div>
                    <label> Guess Entry </label>
                    <textarea />
                </div>
                <div>
                    <button>
                        Sign
                    </button>
                    { "user goes here" && (
                        <button
                            type= "button"
                            // onClick={ () => {
                            //     setUser('')
                            //     setName('')
                            // }}
                        >
                         Not {user} ?
                        </button>
                    )}
                </div>
            </form>
        </>
    )


}