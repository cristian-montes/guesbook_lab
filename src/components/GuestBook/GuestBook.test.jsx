import React from "react";

import { render, screen } from '@testing-library/react';
import GuestBook from './GuestBook';
import { EntryProvider } from "../../context/EntryContext";
import { UserProvider } from "../../context/UserContext";

it('should render the gesrbook', ()=>{
    const { container } = render(
    <UserProvider>
        <EntryProvider>
            < GuestBook />
        </EntryProvider>
    </UserProvider>
    );

    expect(container).toMatchSnapshot();
});
