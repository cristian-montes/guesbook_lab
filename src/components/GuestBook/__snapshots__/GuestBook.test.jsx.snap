// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the gesrbook 1`] = `
<div>
  <form>
    <h2>
      Please Sign the Guestbook
    </h2>
    <label
      for="guestEntry"
    >
      Guest Entry
    </label>
    <textarea
      id="guestEntry"
      placeholder="Your Entry..."
    />
    <button
      type="submit"
    >
      Sign
    </button>
    
  </form>
</div>
`;
